src/app/shell/header.component.js

```diff
@@ -0,0 +1,89 @@
+import template from './header.html';
+import aboutTemplate from 'app/views/about.html';
+import supportTemplate from 'app/views/support.html';
+import filtersTemplate from 'app/views/filters.html';
+import './header.scss';
+
+// TODO: Today we share one header everywhere, and show/hide bits of it depending on the circumstance.
+// It'd be nice if there were a cleaner way to go about this.
+export const HeaderComponent = {
+  template,
+  controller: HeaderController
+};
+
+function HeaderController(
+  dimState,
+  ngDialog,
+  $rootScope,
+  hotkeys,
+  dimXurService,
+  dimSettingsService,
+  $transitions,
+  $state
+) {
+  'ngInject';
+
+  const vm = this;
+
+  // Variables for templates that webpack does not automatically correct.
+  vm.$DIM_VERSION = $DIM_VERSION;
+  vm.$DIM_FLAVOR = $DIM_FLAVOR;
+  vm.$DIM_CHANGELOG = $DIM_CHANGELOG;
+
+  vm.settings = dimSettingsService;
+
+  vm.featureFlags = {
+    vendorsEnabled: $featureFlags.vendorsEnabled,
+    activities: $featureFlags.activities
+  };
+
+  vm.destinyVersion = getCurrentDestinyVersion($state);
+
+  function getCurrentDestinyVersion() {
+    // TODO there must be a better way of doing this?
```

Comment:
something like `return /destiny(\d+)/.exec($state.current.name)[1]`? or do you mean a better way all together